src/libcore/result.rs

```diff
@@ -501,6 +501,8 @@ impl<T, E> Result<T, E> {
 
     /// Returns an iterator over the possibly contained value.
     ///
+    /// The iterator yields one value if the result is [`Ok`], otherwise none.
```

Comment:
Well, no, it doesn't yield `None`, it just yields no values at all.

I realize this is implemented by `next()` returning `None` but I think saying that would only make this confusing.  I could rephrase it as "the iterator's next method returns Some of the inner and then None if the result is `Ok`, otherwise it returns None."  But that seems to be needlessly puncturing the iterator abstraction.

A couple of general comments: 

 - There are lots of references to well-known types such as `None` and `Ok` throughout the code without hyperlinks, indeed even in this very file.  If the goal is that every type be manually linked maybe that should go into `CONTRIBUTING.md` to avoid make that clear.  IMO it is not so necessary to linkify values the reader is highly likely to already understand, and that anyhow are linked from nearby.

 - In my experience repeatedly asking contributors for small incremental changes to a PR that's already an improvement tends to discourage future contributions.  My normal practice now is to accept the change and encourage them to send another PR making additional cleanups, and I think this gives them a better feeling about contributing.